ytmusicapi/auth/headers.py

```diff
@@ -20,16 +20,26 @@ def load_headers_file(auth: str) -> Dict:
     return input_json
 
 
-def prepare_headers(
-    session: requests.Session,
-    proxies: Optional[Dict] = None,
-    input_dict: Optional[CaseInsensitiveDict] = None,
-) -> Dict:
+def prepare_headers(auth_session: requests.Session | YTMusicOAuth,
```

Comment:
I think instead of auth_session it would be much nicer and much more readable to just pass in the extra credential for `YTMusicOAuth`